src/main/java/net/minecraftforge/client/extensions/IForgeFont.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.extensions;
+
+import net.minecraft.client.gui.Font;
+import net.minecraft.network.chat.FormattedText;
+
+/**
+ * Extension interface for {@link Font}.
+ */
+public interface IForgeFont
+{
+    FormattedText ELLIPSIS = FormattedText.of("...");
+
+    private Font self()
```

Comment:
On rendering Renate objects this method needs to be patched into the target instead of in gehe extension interface.

At the moment the jvm will generate bytecode which forces the runtime to check if IForgeFont's this instance is actually an instance of Font before doing the cast, which is really expensive compared to just the cast (several orders of magnitude difference).